src/client/builder.rs

```diff
@@ -128,4 +129,433 @@ impl ClientBuilder {
+    /// |`IntegratedSecurity`|Toggle between Windows authentication and SQL authentication.|
+    /// |`uid`, `username`, `user`|The SQL Server login account.|
+    /// |`password`, `pwd`|The password for the SQL Server account logging on.|
+    /// |`database`|The name of the database.|
+    /// |`TrustServerCertificate`|Specifies whether the driver trusts the server certificate when connecting using TLS.|
+    /// |`encrypt`|Specifies whether the driver uses TLS to encrypt communication.|
+    pub fn from_ado_string(s: &str) -> crate::Result<Self> {
+        let ado = AdoNetString::parse(s)?;
+        let mut builder = Self::default();
+
+        let server = ado.server()?;
+
+        if let Some(host) = server.host {
+            builder.host(host);
+        }
+
+        if let Some(port) = server.port {
+            builder.port(port);
+        }
+
+        if let Some(_instance) = server.instance {
+            #[cfg(windows)]
+            builder.instance_name(_instance);
+        }
+
+        builder.authentication(ado.authentication()?);
+
+        if let Some(database) = ado.database() {
+            builder.database(database);
+        }
+
+        if ado.trust_cert()? {
+            builder.trust_cert();
+        }
+
+        builder.encryption(ado.encrypt()?);
+
+        Ok(builder)
+    }
+}
+
+pub(crate) struct AdoNetString {
+    dict: HashMap<String, String>,
+}
+
+impl AdoNetString {
+    pub fn parse(s: &str) -> crate::Result<Self> {
+        let dict: crate::Result<HashMap<String, String>> = s
+            .split(";")
+            .filter(|kv| kv != &"")
+            .map(|kv| {
+                let mut splitted = kv.split("=");
+
+                let key = splitted
+                    .next()
+                    .ok_or_else(|| {
+                        crate::Error::Conversion(
+                            "Missing a valid key in connection string parameters.".into(),
+                        )
+                    })?
+                    .trim()
+                    .to_lowercase();
+
+                let value = splitted
+                    .next()
+                    .ok_or_else(|| {
+                        crate::Error::Conversion(
+                            "Missing a valid key in connection string parameters.".into(),
+                        )
+                    })?
+                    .trim()
+                    .to_string();
+
+                Ok((key, value))
+            })
+            .collect();
+
+        Ok(Self { dict: dict? })
+    }
+
+    pub fn server(&self) -> crate::Result<ServerDefinition> {
+        fn parse_server(parts: Vec<&str>) -> crate::Result<ServerDefinition> {
+            if parts.is_empty() || parts.len() >= 3 {
+                return Err(crate::Error::Conversion("Server value faulty.".into()));
+            }
+
+            let definition = if parts.len() == 1 {
+                // Connect using a host and an instance name, we first need to resolve to a port
```

Comment:
I don't understand this comment. It looks like the port is hardcoded below.

Reply:
Hmm. Yep. So. We have SQL Browser here. `tcp:host\\INSTANCE`, and this then binds UDP to the default port of the browser. This is kind of taken from the old impl, but you're right, it's fishy.

What should be here is matching if the string has `\\` somewhere instead, taking the possible port value if needed.

Reply:
Fixed in the next commit.